### PR TITLE
Mesh 826/systematic cdd claims

### PR DESCRIPTION
**Pull request by fmiguelgarcia**
This PR covers MESH-788 & MESH-789.
It automatically generates  CDD claims (a.k.a. Systematic CDD Claims) for any Identity which is added to GC or CDD Trusted Providers. Once a member of those groups is removed, that claim is revoked too.

There are two Systematic Claim Issuer, one for each group. I did that to support the edge case of having an Identity in both groups. I added a UT to cover that weird case.

### CODE REVIEW

File: pallets/committee/src/lib.rs [hunk 1]
```diff
@@ -357,11 +357,7 @@ impl<T: Trait<I>, I: Instance> GroupTrait<T::Moment> for Module<T, I> {
 }
 
 impl<T: Trait<I>, I: Instance> ChangeMembers<IdentityId> for Module<T, I> {
-    fn change_members_sorted(
-        _incoming: &[IdentityId],
-        outgoing: &[IdentityId],
-        new: &[IdentityId],
-    ) {
+    fn change_members_sorted(incoming: &[IdentityId], outgoing: &[IdentityId], new: &[IdentityId]) {
```
Comment (maxsam4) [inline pallets/committee/src/lib.rs]: Maybe add a dev note for future that this function specifically uses `SystematicIssuers::GovernanceCommittee` as cdd issuer so if multiple instances of this were used of this module then a different implementation of these functions should be used (like we do in cdd providers).

Comment (fmiguelgarcia) [reply pallets/committee/src/lib.rs]: Comment added. I've also renamed `SystematicIssuers::GovernanceCommittee` to `SystematicIssuers::Committee`.

File: pallets/common/src/constants.rs [hunk 2]
```diff
@@ -43,9 +43,12 @@ pub mod fee {
 /// DID-related.
 pub mod did {
     /// prefix for user dids
-    pub const USER: [u8; 5] = *b"USER:";
+    pub const USER: &[u8; 5] = b"USER:";
     /// prefix for security token dids
-    pub const SECURITY_TOKEN: [u8; 15] = *b"SECURITY_TOKEN:";
+    pub const SECURITY_TOKEN: &[u8; 15] = b"SECURITY_TOKEN:";
+    /// Governan Committee DID. It is used in syste
```
Comment (adamdossa) [inline pallets/common/src/constants.rs]: minor - looks like comment got chopped - please can we add a comment for the CDD_PROVIDERS_ID as well

File: pallets/common/src/lib.rs [hunk 3]
```diff
@@ -10,3 +10,42 @@ pub use context::Context;
...
+/// too.
+///
+/// There is just one Systematic claim at the moment, and It is generated by two issuers.
+/// It is a CDD claim that is assigned to members of specific groups:
+/// * Governance Committee: Each member of the committee has a CDD claim generated by
+/// `SystematicIssuers::Committee` in order to allow it to operate independently of
+/// claims generated by CDD trusted providers. Using that claim, GC members could operate even if
+/// their CDD claim's issuer has been revoked.
+/// * CDD Service Providers: Every CDD providers has a CDD claim generated by
+/// `SystematicIssuers::CDDProvider` group, in order to avoid self-generated claim issue.
```
Comment (adamdossa) [inline pallets/common/src/lib.rs]: Out of interest, is there a technical reason to have a different systematic issuer for CDDs vs. GC members? In my head they are both being issued by "Polymesh" or arguably the governance committee in a systematic sense.

Comment (adamdossa) [reply pallets/common/src/lib.rs]: I guess it means that if you have a CDD provider that is also a GC member, then they would have two claims (one from Committee, one from CDDProvider). If they left one of the two groups, there other claim would remain so I guess it makes that logic simpler which is good.

Comment (fmiguelgarcia) [reply pallets/common/src/lib.rs]: Yes, that is. Actually I added a test case where an Identity is in CDD & GC group at the same time.
Test: `gc_and_cdd_with_systematic_cdd_claims` in `identity_test.rs`

File: pallets/common/src/constants.rs (hunk 2 above)
Comment (fmiguelgarcia) [reply pallets/common/src/constants.rs]: Fixed.

**Pull request merged**